drivers/display/display_sh1122.c

```diff
@@ -0,0 +1,538 @@
+				    const struct display_buffer_descriptor *desc)
+{
+	const struct sh1122_config *config = dev->config;
+	struct display_buffer_descriptor mipi_desc;
+
+	mipi_desc.buf_size = desc->width / 2;
+	mipi_desc.pitch = desc->pitch / 2;
+	mipi_desc.width = desc->width;
+	mipi_desc.height = 1;
+
+	/* This is the wrong format, but it doesn't matter to almost all mipi drivers */
+	return mipi_dbi_write_display(config->mipi_dev, &config->dbi_config, config->conversion_buf,
+				      &mipi_desc, PIXEL_FORMAT_L_8);
+}
+
+static int sh1122_release_bus_mipi(const struct device *dev)
+{
+	const struct sh1122_config *config = dev->config;
+
+	return mipi_dbi_release(config->mipi_dev, &config->dbi_config);
+}
+#endif
+
+#if DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(sinowealth_sh1122, i2c)
+static int sh1122_write_pixels_i2c(const struct device *dev, const uint8_t *buf,
+				   const struct display_buffer_descriptor *desc)
+{
+	const struct sh1122_config *config = dev->config;
+	int ret, i;
+	int total = 0;
+
+	return i2c_burst_write_dt(&config->i2c, SH1122_CONTROL_ALL_BYTES_DATA,
+				  config->conversion_buf, desc->width / 2);
+}
+
+static int sh1122_release_bus_i2c(const struct device *dev)
+{
+	return 0;
+}
+#endif
+
+static int sh1122_write(const struct device *dev, const uint16_t x, const uint16_t y,
+			const struct display_buffer_descriptor *desc, const void *buf)
+{
+	const struct sh1122_config *config = dev->config;
+	int err;
+	size_t buf_len;
+	int pixel_count = desc->width * desc->height;
+	int total = 0;
+	uint8_t ybuf = y;
+
+	if (desc->pitch < desc->width) {
+		LOG_ERR("Pitch is smaller than width");
+		return -1;
+	}
+
+	/* Following the datasheet, in the GDDRAM, two segment are split in one register */
+	buf_len = MIN(desc->buf_size, desc->height * desc->width / 2);
+	if (buf == NULL || buf_len == 0U) {
+		LOG_ERR("Display buffer is not available");
+		return -1;
+	}
+
+	if (desc->pitch > desc->width) {
+		LOG_ERR("Unsupported mode");
+		return -1;
+	}
+
+	if ((x & 1) != 0U) {
+		LOG_ERR("Unsupported origin");
+		return -1;
+	}
+
+	LOG_DBG("x %u, y %u, pitch %u, width %u, height %u, buf_len %u", x, y, desc->pitch,
+		desc->width, desc->height, buf_len);
+
+	while (pixel_count > total) {
+
+		err = config->write_cmd(dev, SH1122_SET_COLUMN_ADDR_HIGH(x / 2), NULL, 0);
+		if (err) {
+			return err;
+		}
+
+		err = config->write_cmd(dev, SH1122_SET_COLUMN_ADDR_LOW(x / 2), NULL, 0);
+		if (err) {
+			return err;
+		}
+
+		err = config->write_cmd(dev, SH1122_SET_ROW_ADDR, &ybuf, 1);
+		if (err) {
+			return err;
+		}
+
+		for (int i = 0; desc->width > i; i += 2) {
+			config->conversion_buf[i / 2] = ((uint8_t *)buf)[total + i + 1] >> 4;
+			config->conversion_buf[i / 2] |= (((uint8_t *)buf)[total + i] >> 4) << 4;
+		}
+
+		err = config->write_pixels(dev, config->conversion_buf, desc);
+		if (err) {
+			return err;
+		}
+		total += desc->width;
+		ybuf++;
+	}
+	config->release_bus(dev);
+	return 0;
```

Comment:
return the result of `release_bus` instead.

Reply:
The meaning here would be that we dont care about the return value of release_bus

Reply:
Done for consistency since I havent ignored it for other rest, doesnt really matter in the end.